quote-server/server.js

```diff
@@ -1,32 +1,59 @@
-// server.js
-// This is where your node app starts
-
-//load the 'express' module which makes writing webservers easy
 import express from "express";
-//load the quotes JSON
-import quotes from "./quotes.json" assert { type: "json" };
+import crypto from "crypto";
+import cors from "cors";
 
 const app = express();
-// Now register handlers for some routes:
-//   /                  - Return some helpful welcome info (text)
-//   /quotes            - Should return all quotes (json)
-//   /quotes/random     - Should return ONE quote (json)
-app.get("/", (request, response) => {
-  response.send("Neill's Quote Server!  Ask me for /quotes/random, or /quotes");
+
+let quotes = [];
+
+app.use(cors());
+
+app.use(async (req, res, next) => {
+  const response = await fetch("https://api.quotable.io/quotes?limit=150");
+  const data = await response.json();
+  quotes = data.results;
+  next();
+});
+
+app.get("/", (req, res) => {
+  res.send(
+    "Fikret's Quote Server! Ask me for /quotes, /quotes/random or /quotes/search?term=(something)"
+  );
 });
 
 //START OF YOUR CODE...
+app.get("/quotes", (req, res) => {
+  res.json({ quotes });
+});
 
+app.get("/quotes/random", (req, res) => {
+  const randomQuote = pickFromArray(quotes);
+  res.json({ randomQuote });
+});
+
+app.get("/quotes/search", (req, res) => {
+  console.log(req.query);
+  const searchTerm = req.query.term.toLowerCase();
+
+  const filteredQuotes = quotes.filter(
+    (quoteObject) =>
+      quoteObject.content.toLowerCase().includes(searchTerm) ||
+      quoteObject.author.toLowerCase().includes(searchTerm)
+  );
+  res.json({ filteredQuotes });
+});
+
+app.get("/echo", (req, res) => {
+  res.json(req.query.word);
+});
 //...END OF YOUR CODE
 
-//You can use this function to pick one element at random from a given array
-//example: pickFromArray([1,2,3,4]), or
-//example: pickFromArray(myContactsArray)
-//
-const pickFromArray = (arrayofQuotes) =>
-  arrayofQuotes[Math.floor(Math.random() * arrayofQuotes.length)];
+const pickFromArray = (arrayOfQuotes) => {
```

Comment:
What's the difference between this implementation of `pickFromArray` and the one that was supplied? What do you prefer about this one?

Reply:
Math.random() generates a random number between 0 and 1. this is not cryptographically random and it was causing error in the code checking bot. I used crypto to give me bytes and converted it to integer and than used the  reminder from length of the array as index number to get more secure randomness. This way I was able to pass the bot :)

Reply:
Aha! I think this is a case of an over-zealous bot :) I think the code was fine before, and the code is fine after too, but for the purposes of picking a random quote from a list you probably don't need cryptographically secure randomness!

Reply:
Haha yes there is no need. But when there is a failed test it really annoys me 😂 plus I learned how to get randomness more secure 🤪